Rubberduck.CodeAnalysis/Inspections/Concrete/UnreachableCaseInspection/ExpressionFilter.cs

```diff
@@ -0,0 +1,628 @@
+            {
+                range = TrimEnd(range, Limits.Maximum);
+            }
+
+            if (!Ranges.Any())
+            {
+                Ranges.Add(range);
+                return true;
+            }
+            else
+            {
+                var initial = Ranges.Count;
+                RemoveRangesCoveredByRange(range);
+
+                if (!TryMergeWithOverlappingRange(range))
+                {
+                    Ranges.Add(range);
+                }
+                return initial != Ranges.Count;
+            }
+        }
+
+        protected bool FiltersTrueFalse => FiltersValue(_trueValue) && FiltersValue(_falseValue);
+
+        protected virtual bool AddIsClause(IsClauseExpression expression)
+        {
+            if (Converter(expression.LHS, out T value, _filterTypeName))
+            {
+                IsDirty = true;
+                if (IsClauseAdders.ContainsKey(expression.OpSymbol))
+                {
+                    if (IsClauseAdders[expression.OpSymbol](this, value))
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+            else
+            {
+                return AddToContainer(Variables[VariableClauseTypes.Is], expression.ToString());
+            }
+        }
+
+        protected virtual bool AddMinimum(T value)
+        {
+            IsDirty = true;
+            var result = Limits.SetMinimum(value);
+            if (TryGetMinimum(out T min))
+            {
+                var newRanges = new HashSet<(T Start, T End)>();
+                foreach ( var range in Ranges)
+                {
+                    newRanges.Add(TrimStart(range, min));
+                }
+                Ranges = newRanges;
+
+                SingleValues.Where(sv => sv.CompareTo(min) < 0).ToList()
+                    .ForEach(sv => SingleValues.Remove(sv));
+            }
+            return result;
+        }
+
+        protected virtual bool AddMaximum(T value)
+        {
+            IsDirty = true;
+            var result =  Limits.SetMaximum(value);
+            if (TryGetMaximum(out T max))
+            {
+                var newRanges = new HashSet<(T Start, T End)>();
+                foreach (var range in Ranges)
+                {
+                    newRanges.Add(TrimEnd(range, max));
+                }
+                Ranges = newRanges;
+
+                SingleValues.Where(sv => sv.CompareTo(max) > 0).ToList()
+                    .ForEach(sv => SingleValues.Remove(sv));
+            }
+            return result;
+        }
+
+        protected void RemoveRangesCoveredByLimits()
+        {
+            var rangesToRemove = Ranges.Where(rg => Limits.CoversRange(rg));
+            foreach(var range in rangesToRemove)
+            {
+                Ranges.Remove(range);
+            }
+
+        }
+
+        protected void RemoveRangesCoveredByRange((T Start, T End) range)
```

Comment:
Given that we use the `(T Start, T End)` a lot and not just in private, I think it's better to define them as a proper `struct` or `class` object and reference to that instead. 

Furthermore, that would allow you to separate out the operations on the `(T Start, T End)` as its own methods, apart from this class, which would make it more readable and understandable.

Reply:
Replaced `Tuple <T,T>(T Start, T End)` for ranges as RangeOfValues struct....I agree, the change does make the code more readable.